### PR TITLE
#138: Make a universal forwarder part of a host image

### LINKED ISSUE #138: Dynamic host discovery
We have a use case in which the host on which we configure a Splunk forwarder (using this cookbook) is different than the host where it's eventually deployed. 

In order to associate the logs with the correct host, we'd like to submit a request to add the ability to dynamically inject current hostname into the forwarder configuration (i.e. inputs.conf) whenever splunkd starts or restarts.

### PR DESCRIPTION
**Pull request by jeffzjd**
Fixes #138

### CODE REVIEW

File: recipes/clone_prep_clear_config.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# coding: UTF-8
+#
+# Cookbook Name:: cerner_splunk
+# Recipe:: clone_prep_clear_config
+#
+# Make a universal forwarder part of a host image.
+#
+# See http://docs.splunk.com/Documentation/Forwarder/latest/Forwarder/Makeauniversalforwarderpartofahostimage
```
Comment (acharlieh) [inline recipes/clone_prep_clear_config.rb]: Change latest to 6.5.2. It has been known that pages change names between versions, and latest is always the latest version (and pages happen to change). 

Possibly also add a comment how this recipe requires Chef 12.6.0 or greater. (That's where the `:before` notification timing was introduced).

File: recipes/clone_prep_clear_config.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+execute 'clone-prep-clear-config' do
+  command "#{node['splunk']['cmd']} clone-prep-clear-config"
+  user node['splunk']['user']
+  group node['splunk']['group']
+  only_if { ::File.exist? "#{node['splunk']['cmd']}" }
```
Comment (acharlieh) [inline recipes/clone_prep_clear_config.rb]: I don't know that we want to guard this execution. If someone has configuration that for whatever reason the splunk cmd does not exist by this point, we would want chef to blow up.

File: README.md [hunk 3]
```diff
@@ -96,6 +96,9 @@ Possibly Asked Questions
 * Do I have to specify an index?
     * At Cerner: Yes. We use indexes to define ownership and access to data. Data sent to the default index will be rejected.
     * In general: No. Without specifying an index, you wind up in the 'main' index.
+* Can I use this cookbook to configure a Universal Forwarder for a host image?
+    * Yes, add `recipe[cerner_splunk::image_prep]` to the end of your run list.
+
```
Comment (acharlieh) [inline README.md]: Nit extra line... 

But also could you squash your commits, and stop referencing the issue from your commit message? (as is stated in https://github.com/cerner/cerner_splunk/blob/stable/CONTRIBUTING.md#feature-develoment ) as you can see it's adding noise to the issue

**Comment by jeffzjd** [comment]
@acharlieh I removed the extra line and squashed the commits.

**Comment by nmattam** [comment]
+1

**Comment by gravesb** [comment]
+1

**Comment by alecxvs** [comment]
+1

**Comment by acharlieh** [comment]
This was merged with 66e8834d5c1d8feb95c26bdd2657dc97ebfe9683 to stable. (This PR is against the wrong branch.

**Pull request closed without merging**